apispec/ext/marshmallow/swagger.py

```diff
@@ -291,13 +292,14 @@ class Meta:
     if getattr(Meta, 'fields', None) or getattr(Meta, 'additional', None):
         warnings.warn('Only explicitly-declared fields will be included in the Schema Object. '
                 'Fields defined in Meta.fields or Meta.additional are excluded.')
-    ret = {'properties': {}}
+    ret = {'properties': LazyDict({})}
     exclude = set(getattr(Meta, 'exclude', []))
     for field_name, field_obj in iteritems(fields):
         if field_name in exclude or (field_obj.dump_only and not dump):
             continue
-        prop = field2property(field_obj, spec=spec, use_refs=use_refs, dump=dump)
-        ret['properties'][field_name] = prop
+        propFunc = lambda field_obj=field_obj:\
```

Comment:
> What does the assignment in the lambda do?

it creates a closure around the `spec` variable. read [here](http://stackoverflow.com/questions/2295290/what-do-lambda-function-closures-capture-in-python/2295372#2295372)